### PR TITLE
Single step deploy

### PR DESCRIPTION
**Pull request by davidegaspar**

### CODE REVIEW

File: cfn/services/cloudformation.py [hunk 1]
```diff
@@ -45,39 +54,60 @@ def validate_template_config(config):
...
             args.kms_key_id,
             False
         )
         template = Template(args.template, os.getcwd(), s3_uploader)
         exported_template = template.export()
         exported_template_yaml = yaml_dump(exported_template)
     except exceptions.ExportFailedError as ex:
-        logging.error(ex)
+        if template_has_resources_to_upload_to_s3(template) and not args.s3_bucket:
+            logging.error('The template contains resources to export, please provide an S3 Bucket (--s3-bucket).')
```
Comment (cfbarbero) [inline cfn/services/cloudformation.py]: As we discussed, 'export' is a misleading term here.  
Please change this to say something like "the template contains resources to upload, ..."

File: cfn/subcommands/deploy.py [hunk 2]
```diff
@@ -31,34 +30,58 @@ def add_subparser(subparsers):
...
     parser.add_argument(
         '--kms-key-id',
         '-k',
         help='The ID of an AWS KMS key that the command uses to encrypt artifacts that are at rest in the S3 bucket.'
     )
+    parser.add_argument(
+        '--output-template-file',
+        '-o',
+        help='The path to the file where the command writes the output AWS CloudFormation template.',
+        default='packaged.yml'
```
Comment (cfbarbero) [inline cfn/subcommands/deploy.py]: add the default to the help text

File: cfn/subcommands/deploy_only.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    parser.add_argument(
+        '--profile',
+        '-p',
+        help='aws profile'
+    )
+    parser.add_argument(
+        '--packaged',
+        '-k',
+        help='The path where your AWS CloudFormation packaged template is located.',
+        default='packaged.yml'
```
Comment (cfbarbero) [inline cfn/subcommands/deploy_only.py]: add the default to the help text

File: cfn/subcommands/package.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    parser.add_argument(
+        '--kms-key-id',
+        '-k',
+        help='The ID of an AWS KMS key that the command uses to encrypt artifacts that are at rest in the S3 bucket.'
+    )
+    parser.add_argument(
+        '--output-template-file',
+        '-o',
+        help='The path to the file where the command writes the output AWS CloudFormation template.',
+        default='packaged.yml'
```
Comment (cfbarbero) [inline cfn/subcommands/package.py]: add the default to the help text

File: cfn/services/cloudformation.py (hunk 1 above)
Comment (davidegaspar) [reply cfn/services/cloudformation.py]: updated to template_has_resources_to_upload_to_s3

File: cfn/subcommands/deploy.py (hunk 2 above)
Comment (davidegaspar) [reply cfn/subcommands/deploy.py]: done

File: cfn/subcommands/deploy_only.py (hunk 3 above)
Comment (davidegaspar) [reply cfn/subcommands/deploy_only.py]: done

File: cfn/subcommands/package.py (hunk 4 above)
Comment (davidegaspar) [reply cfn/subcommands/package.py]: done

File: cfn/version.py [hunk 5]
```diff
@@ -1 +1 @@
-__version__ = '1.0.1'
```
[the remark this answers is not in the document]
Comment (davidegaspar) [reply cfn/version.py]: old

**Comment by davidegaspar** [comment]
all comments have been addressed

**Comment by cfbarbero** [comment]
Please create a release for this when you merge so that people can install specific versions of the tool.

**Pull request merged**